src/languageservice/serviceclient.ts

```diff
@@ -191,7 +191,10 @@ export default class SqlToolsServiceClient {
                         });
                     } else {
                         this.initializeLanguageClient(serverPath, context);
-                        resolve(new ServerInitializationResult(false, true, serverPath));
+
+                        this._client.onReady().then(() => {
```

Comment:
Can we catch the failure case here? Or else return this .then from so that the outer .catch will be called?
- Recommend changing to async for clarity on this. Inner thenables have this problem, e.g. what are we returning?

Reply:
Basically because we don't return `this._client.onReady().then` we risk having unhandled promise instead of the outer .catch on the `this._server.getServerPath(platformInfo.runtimeId).then` being called.

Reply:
I switched to await, so I think the exception would go to the existing .catch block below.  Please let me know if this doesn't address the concern.